### PR TITLE
Add missing arguments

### LINKED ISSUE #70: Ability to configure external caches for thanos store
**Is your feature request related to a problem? Please describe.**

We use memcached as an external cache for thanos-store, for both the index and chunk caches. See `--index-cache.config` and `--store.caching-bucket.config` in https://thanos.io/tip/components/store.md.

**Describe the solution you'd like to see**

Ideally, there would be configurable fields under `Thanos.spec.storeGateway` or `StoreEndpoint.spec` that translate into the YAML snippets that Thanos uses as external cache configs. These could be passed as string literals into flags, or mounted as files from ConfigMaps or Secrets.

**Describe alternatives you've considered**

Alternatively / in addition, we could add an "extra args" setting to allow users to pass arbitrary flags to thanos store (and other thanos components). It looks like this could either be done in this repo or by adding an "Args" field to ContainerBase in https://github.com/banzaicloud/operator-tools.

**Additional context**

I'm happy to send a PR, and I'm of course interested in what the maintainers think about the problem, and its proposed solutions.

### LINKED ISSUE #72: thanos-store flags: index cache postings compression and relabel configs
**Is your feature request related to a problem? Please describe.**

We'd like to set the following flags on the thanos store component:

- `--experimental.enable-index-cache-postings-compression`
- `--selector.relabel-config` / `--selector.relabel-config-file`

**Describe the solution you'd like to see**

We could add these options to Thanos.spec.storeGateway.

**Describe alternatives you've considered**

While no substitute for spec fields, https://github.com/banzaicloud/thanos-operator/pull/71 could be a nice escape hatch allowing users to set arbitrary thanos (store) flags before operator support is added.

**Additional context**

I'm happy to send PRs for this, but it'd be nice to get feedback on https://github.com/banzaicloud/thanos-operator/pull/71 first.

### PR DESCRIPTION
**Pull request by tarokkk**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #70 #72
| License         | Apache 2.0

New Thanos versions introduced new arguments that we haven't supported. This PR adds these to the components.

### CODE REVIEW

**Review by bonifaido — APPROVED** [review]

File: pkg/sdk/api/v1alpha1/thanos_types.go [hunk 1]
```diff
@@ -239,18 +239,61 @@ type StoreGateway struct {
 	GRPCGracePeriod string `json:"grpcGracePeriod,omitempty" thanos:"--grpc-grace-period=%s"`
 	// Maximum size of items held in the in-memory index cache.
 	IndexCacheSize string `json:"indexCacheSize,omitempty" thanos:"--index-cache-size=%s"`
+	// Path to YAML file that contains index cache configuration. See format details:
+	// https://thanos.io/tip/components/store.md/#index-cache
+	IndexCacheConfigFile string `json:"indexCacheConfigFile" thanos:"index-cache.config-file=%s"`
+	// Alternative to 'index-cache.config-file' flag (lower priority). Content of YAML file that contains index cache configuration. See format details:
+	// https://thanos.io/tip/components/store.md/#index-cache
+	IndexCacheConfig string `json:"indexCacheConfig" thanos"--index-cache.config=%s"`
```
Comment (pepov) [inline pkg/sdk/api/v1alpha1/thanos_types.go:247]: Are these really required or you just left out the omitempty?

File: pkg/sdk/api/v1alpha1/thanos_types.go [hunk 2]
```diff
@@ -239,18 +239,61 @@ type StoreGateway struct {
...
 	// Maximum size of concurrently allocatable bytes for chunks.
 	ChunkPoolSize string `json:"chunkPoolSize,omitempty" thanos:"--chunk-pool-size=%s"`
 	// Maximum amount of samples returned via a single Series call. 0 means no limit. NOTE: For
 	// efficiency we take 120 as the number of samples in chunk (it cannot be bigger than that), so
 	// the actual number of samples might be lower, even though the maximum could be hit.
 	StoreGRPCSeriesSampleLimit string `json:"storeGRPCSeriesSampleLimit,omitempty" thanos:"--store.grpc.series-sample-limit=%s"`
+	// Maximum amount of touched series returned via a single Series call. The Series call fails if this limit is exceeded. 0 means no limit.
+	StoreGRPCTouchedSeriesSampleLimit int `json:"storeGRPCTouchedSeriesSampleLimit" thanos:"--store.grpc.touched-series-limit=%d"`
```
Comment (pepov) [inline pkg/sdk/api/v1alpha1/thanos_types.go:255]: Same as above, is this required?

**Review by pepov — COMMENTED** [review]
I've added two comments, but didn't want to add for all fields, just please double check whether omitempty is missing from the new fields or not and if missing maybe some of those need to be pointers to allow 0 or empty string as well?

File: pkg/sdk/api/v1alpha1/thanos_types.go (hunk 1 above)
Comment (tarokkk) [reply pkg/sdk/api/v1alpha1/thanos_types.go]: Nice catch thank you!

File: pkg/sdk/api/v1alpha1/thanos_types.go (hunk 2 above)
Comment (tarokkk) [reply pkg/sdk/api/v1alpha1/thanos_types.go]: Nice catch thank you!

**Pull request merged**